tests/mock_code/test_isolated.py

```diff
@@ -61,3 +61,29 @@ def test_failure_no_executable(pytester: pytest.Pytester):
     pytester.makepyfile(PYFILE)
     result = pytester.runpytest_subprocess("-k", "test_basic")
     result.stdout.re_match_lines([r".*ERROR: No existing cache, and no executable specified.*"])
+
+
+def test_custom_hasher(pytester: pytest.Pytester):
+    """Test using a custom hasher."""
+    pytester.makeconftest(CONFTEST)
+    pytester.path.joinpath("file1.txt").write_text("a")
+    pytester.path.joinpath("file2.txt").write_text("b")
+    pytester.makepyfile(
+        """
+        from aiida.engine import run_get_node
+        from aiida_testing.mock_code import InputHasher
+        class CustomHasher(InputHasher):
+            def modify_content(self, path, content):
+                if path.name == 'file1.txt':
+                    self.log("Skipping file1.txt")
+                    return None
+                return content
+        def test_basic(mock_code_factory, generate_diff_inputs):
+            mock_code = mock_code_factory('diff', hasher=CustomHasher)
+            builder = mock_code.get_builder()
+            run_get_node(builder, **generate_diff_inputs())
+            assert False
```

Comment:
What is this line doing here?

Reply:
Making this test fail, so that pytest outputs the full "log" for it, in this case primarily tue captured stdout with the mock code log in it. If the test fails you don't see this.